cmd/util/hook/version.go

```diff
@@ -19,13 +19,7 @@ func StartUpdateCheck(cmd *cobra.Command, args []string) {
 	version.RunUpdateChecker(
 		cmd.Context(),
 		func(ctx context.Context) (*models.BuildVersionInfo, error) {
-			if response, err := util.GetAPIClientV2(cmd).Agent().Version(ctx); err != nil {
-				return nil, err
-			} else if response != nil {
-				return response.BuildVersionInfo, nil
-			} else {
-				return nil, nil
-			}
+			return version.Get(), nil
```

Comment:
This is the fix for: https://github.com/bacalhau-project/bacalhau/issues/3773:

This is because the update check runs async on every command and incorrectly required a client, which prompted for a token to ask the server what version is was**. The actual command being run (e.g. `job run`) also created a client and asked for a token which is why the CLI asked for a token twice until and `update.json` file existed. The fix here is to simply return the version of the binary running the update checker.

** no idea why we do this, update checker should tell you the binary you are using is out of date, not the server you are talking to. Server operators will become aware their server is out of date via log message the update server prints when its update checker runs

Reply:
maybe we need to understand the purpose of the update checker better. Is it to tell that the client is outdated compared to the orchestrator's version? or outdated compared to latest releases from bacalhau?

As a user, I _might_ expect the following:
1. If I am running a compute node, I would like the cli to compare my version with the orchestrator's version. I don't want the cli to ask me to update to a version that is beyond the orchestrator's
2. If I am running client commands, such as `bacalhau job run`, I would like the cli to also compare my version with the orchestrator's
3. If I a m running an orchestrator node, then I would like the cli to compare my version with the latest release from bacalhau

@aronchick @MichaelHoepler what do you think the right behaviour here?

Reply:
I'd propose we reduce the responsibility of the UpdateChecker to only retrieve the latest version of bacalhau from the update server. This will remove the dependency on a bacalhau client - and thus authentication - from the UpdateChecker service - which will fix this bug.

The UpdateChecker may then persist the value it retrieves from the update server to some "system configuration field" (will be very easy to do when [#3959](https://github.com/bacalhau-project/bacalhau/pull/3959/files#diff-a3d824da3c42420cd5cbb0a4a2c0e7b5bfddd819652788a0596d195dc6e31fa5R71) lands). From there we can make the following changes:
1. Clients and Compute nodes can compare their version, the orchestrator's version, and the latest version notifying accordingly in stderr or log messages.
2. Orchestrator nodes can compare their version with the latest version notifying accordingly in log messages.

----------------------------------------------------------

Alternatively, and this would be my preference: 
1. Only Servers (Compute and Requester Nodes) use the update checker (as we already do now) to retrieve the latest version. If they are behind it they produce a log message stating so.
2. Clients are only notified when the server they are communicating with is on a version different than theirs. Clients commands do not run the UpdateChecked.

In this flow, clients are only notified they need to update when their server operator updates.

Reply:
Yeah those are valid points. I've approved the PR as figuring out the right behaviour of the update checker is a bigger task, and you are not changing the behaviour with this PR
https://github.com/bacalhau-project/bacalhau/issues/3980